### PR TITLE
fix: increment `vnode._original` to make sure the component is re-ren…

### PR DESCRIPTION
**Pull request by HuJean**
#2125 

## Summary by CodeRabbit

* **Bug Fixes**
  * Ensure components reliably re-render when global properties are updated in edge cases where previous updates could be skipped.

* **Tests**
  * Expanded lifecycle tests to cover more update scenarios, hydration flows, and propagation of global prop changes.

* **Changelog**
  * Added a patch release entry documenting the behavioral fix.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).
- [ ] Changeset added, and when a BREAKING CHANGE occurs, it needs to be clearly marked (or not required).

### CODE REVIEW

File: packages/react/runtime/src/lynx/runWithForce.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ export function runWithForce(cb: () => void): void {
 
     const c = oldVNode[COMPONENT];
     if (c) {
+      if (vnode[ORIGINAL] != null && oldVNode[ORIGINAL] === vnode[ORIGINAL]) {
```
Comment (upupming) [inline packages/react/runtime/src/lynx/runWithForce.ts:34]:
```suggestion
      if (vnode !=== oldVNode && vnode[ORIGINAL] != null && oldVNode[ORIGINAL] === vnode[ORIGINAL]) {
```

Comment (upupming) [reply packages/react/runtime/src/lynx/runWithForce.ts:34]: I think we can remove line 11-22 if we do this.

https://github.com/lynx-family/lynx-stack/blob/62ef34ddf7168c8fff8bf2200bbad200ae82e7c6/packages/react/runtime/src/lynx/runWithForce.ts#L11-L22
